controllers/kserve_raw_inferenceservice_controller_test.go

```diff
@@ -0,0 +1,95 @@
+package controllers
+
+import (
+	"context"
+	"fmt"
+	. "github.com/onsi/ginkgo"
+	. "github.com/onsi/gomega"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"time"
+
+	kservev1alpha1 "github.com/kserve/kserve/pkg/apis/serving/v1alpha1"
+	kservev1beta1 "github.com/kserve/kserve/pkg/apis/serving/v1beta1"
+	corev1 "k8s.io/api/core/v1"
+	apierrs "k8s.io/apimachinery/pkg/api/errors"
+)
+
+const (
+	KserveSA = "kserve-sa"
+)
+
+var _ = Describe("The KServe Dashboard reconciler", func() {
+	var testNs string
+
+	createServingRuntime := func(namespace, path string) *kservev1alpha1.ServingRuntime {
+		servingRuntime := &kservev1alpha1.ServingRuntime{}
+		err := convertToStructuredResource(path, servingRuntime)
+		Expect(err).NotTo(HaveOccurred())
+		servingRuntime.SetNamespace(namespace)
+		if err := cli.Create(ctx, servingRuntime); err != nil && !apierrs.IsAlreadyExists(err) {
+			Fail(err.Error())
+		}
+		return servingRuntime
+	}
+
+	createInferenceService := func(namespace, name string, path string) *kservev1beta1.InferenceService {
+		inferenceService := &kservev1beta1.InferenceService{}
+		err := convertToStructuredResource(path, inferenceService)
+		Expect(err).NotTo(HaveOccurred())
+		inferenceService.SetNamespace(namespace)
+		if len(name) != 0 {
+			inferenceService.Name = name
+		}
+		inferenceService.Annotations = map[string]string{}
+		inferenceService.Annotations["serving.kserve.io/deploymentMode"] = "RawDeployment"
+		if err := cli.Create(ctx, inferenceService); err != nil && !apierrs.IsAlreadyExists(err) {
+			Fail(err.Error())
+		}
+		return inferenceService
+	}
+
+	BeforeEach(func() {
+		testNs = Namespaces.Create(cli).Name
+
+		inferenceServiceConfig := &corev1.ConfigMap{}
+		Expect(convertToStructuredResource(InferenceServiceConfigPath1, inferenceServiceConfig)).To(Succeed())
+		if err := cli.Create(ctx, inferenceServiceConfig); err != nil && !apierrs.IsAlreadyExists(err) {
+			Fail(err.Error())
+		}
+
+	})
+
+	When("deploying a Kserve RawDeployment model", func() {
```

Comment:
Maybe add a test to check if the service accounts and rolebindings are deleted, wdyt?

Reply:
added